android/src/main/java/com/zmxv/RNSound/RNSoundModule.java

```diff
@@ -87,7 +88,13 @@ public synchronized boolean onError(MediaPlayer mp, int what, int extra) {
         return true;
       }
     });
-    player.prepareAsync();
+
+    try {
+      player.prepareAsync();
+    } catch (IllegalStateException ignored) {
+      // When loading files from a file, we useMediaPlayer.create, which actually
```

Comment:
AFAIK that exception is only thrown if the preparation has already been done. Actual errors should still happen.